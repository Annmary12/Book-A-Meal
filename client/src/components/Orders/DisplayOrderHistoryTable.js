import React from 'react';
import PropTypes from 'prop-types';
import Loader from '../Loader.js';
import Table from '../ResponsiveTable/Table.js';
import Cell from '../ResponsiveTable/Cell.js';
import Row from '../ResponsiveTable/Row.js';
import OrderRow from './OrderRow.js';

const DisplayOrderHistory = ({
  isFetching,
  orders,
  cancelOrder,
  updateOrder
}) => (
  <div>
    <Table>
      <Row header>
        <Cell>Date</Cell>
        <Cell>Meal</Cell>
        <Cell>Qty</Cell>
        <Cell>Price</Cell>
        <Cell>Delivery address</Cell>
        <Cell>Status</Cell>
        <Cell>Actions</Cell>
      </Row>
      {
        !isFetching &&
        orders.map(order => (
          <OrderRow
            key={order.id}
            order={order}
            handleCancelOrder={cancelOrder}
            handleUpdateOrder={updateOrder}
          />
        ))
      }
    </Table>
    {isFetching && <Loader />}
  </div>
);

DisplayOrderHistory.propTypes = {
  orders: PropTypes.arrayOf(PropTypes.object).isRequired,
  isFetching: PropTypes.bool.isRequired,
  cancelOrder: PropTypes.func.isRequired,
  updateOrder: PropTypes.func.isRequired
};

export default DisplayOrderHistory;
