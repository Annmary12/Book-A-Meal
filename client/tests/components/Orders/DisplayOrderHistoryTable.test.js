import React from 'react';
import DisplayOrderHistoryTable from '../../../src/components/Orders/DisplayOrderHistoryTable';

const props = {
  orders: [],
  isFetching: true,
  cancelOrder: jest.fn(),
  updateOrder: jest.fn()
};

describe('DisplayOrderHistoryTable component', () => {
  it('should match snapshot if isFetching is true', () => {
    const wrapper = shallow(
      <DisplayOrderHistoryTable {...props} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot if isFetching is false', () => {
    const wrapper = shallow(
      <DisplayOrderHistoryTable
        {...props}
        isFetching={false}
        orders={[{
          id: '32rdsfafase'
        }]}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});