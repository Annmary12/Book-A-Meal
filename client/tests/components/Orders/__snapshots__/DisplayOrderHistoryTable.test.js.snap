// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayOrderHistoryTable component should match snapshot if isFetching is false 1`] = `
<div>
  <Table>
    <Row
      header={true}
    >
      <Cell
        title=""
      >
        Date
      </Cell>
      <Cell
        title=""
      >
        Meal
      </Cell>
      <Cell
        title=""
      >
        Qty
      </Cell>
      <Cell
        title=""
      >
        Price
      </Cell>
      <Cell
        title=""
      >
        Delivery address
      </Cell>
      <Cell
        title=""
      >
        Status
      </Cell>
      <Cell
        title=""
      >
        Actions
      </Cell>
    </Row>
    <OrderRow
      handleCancelOrder={[MockFunction]}
      handleUpdateOrder={[MockFunction]}
      key="32rdsfafase"
      order={
        Object {
          "id": "32rdsfafase",
        }
      }
    />
  </Table>
</div>
`;

exports[`DisplayOrderHistoryTable component should match snapshot if isFetching is true 1`] = `
<div>
  <Table>
    <Row
      header={true}
    >
      <Cell
        title=""
      >
        Date
      </Cell>
      <Cell
        title=""
      >
        Meal
      </Cell>
      <Cell
        title=""
      >
        Qty
      </Cell>
      <Cell
        title=""
      >
        Price
      </Cell>
      <Cell
        title=""
      >
        Delivery address
      </Cell>
      <Cell
        title=""
      >
        Status
      </Cell>
      <Cell
        title=""
      >
        Actions
      </Cell>
    </Row>
  </Table>
  <Loader
    message="loading..."
    width={40}
  />
</div>
`;
